src/pet.js

```diff
@@ -0,0 +1,73 @@
+const max_fitness = 10;
+const empty_belly = 0;
+const grow_old = 1;
+
+function Pet (name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = empty_belly;
+    this.fitness = max_fitness;
+    this.child = [];
+    };
+
+    Pet.prototype = {
+        get isAlive(){
+            return this.age < 30 && this.hunger < 10 && this.fitness > 0
+        }
+    };
+    
+    Pet.prototype.growUp = function (){
+        if (!this.isAlive){
+            throw new Error('Your pet is no longer alive')
+        }
+        this.age += grow_old;
+    };
+    Pet.prototype.getHungry = function (){ 
+        this.hunger += 5;
+    };
+    Pet.prototype.becomeUnfit = function (){
+        this.fitness -= 3;
+    };
+    Pet.prototype.walk =  function (){
+        if (!this.isAlive){
+            throw new Error('Your pet is no longer alive')
+        }
+        if((this.fitness + 4) <= max_fitness){
+            this.fitness += 4
+        } else {
+            this.fitness = max_fitness
+        }
+    };
+    Pet.prototype.feed = function (){
+        if (!this.isAlive){
+            throw new Error('Your pet is no longer alive')
+        }
+        if((this.hunger + 3) <= empty_belly){
+            this.hunger += 3
+        } else {
+            this.hunger = empty_belly;
+        }
```

Comment:
As above this could be refactored to use the built in JS Math function:
```
    this.hunger = Math.max(this.hunger - 3, empty_belly);
```
At there is a bit of a flaw in how this works right now too. Unless the initial value for hunger is below 0, it will always go to the else condition and reset the hunger value to 0 instead of incrementing it by 3 as would usually be expected. This means it will always default to 0, see the test for how you could have caught this error